lib/Sema/TypeCheckConcurrency.cpp

```diff
@@ -846,8 +900,11 @@ namespace {
         return true;
 
       case ActorIsolation::GlobalActor: {
-        // If the global actor types are the same, we're done.
-        if (contextIsolation.getGlobalActor()->isEqual(globalActor))
+        // If the global actor types are the same, then we're ok and stop there.
+        // Otherwise, we check if it this reference qualifies it as being
+        // part of an implicitly async call.
+        if (contextIsolation.getGlobalActor()->isEqual(globalActor)
```

Comment:
Also really minor, but these conditions are semantically very different. Can they be in different "if" statements to make that clear?